src/operators.jl

```diff
@@ -136,3 +147,7 @@ function inducedsubgraph{T<:AbstractGraph}(g::T, iter)
     end
     return h, newvid
 end
+
+# dispatch for g[[1,2,3]], g[1:3], g[Set([1,2,3])]
+# these are the only allowed dispatches, everything else is slow
+getindex(g::AbstractGraph, iter) = first(induced_subgraph(g, iter))
```

Comment:
We're going to be doing this in the `bfsdfs` branch anyway. My reasoning is because we're passing in an iter which is the forward mapping anyway, so it's known; getting the reverse mapping is as easy as
```

d = Dict{Int,Int}()
for (v,k) in enumerate(iter)
  d[k] = v
end
d
```